### ISSUE TITLE
Issue with fastool when parsing compressed files

### ISSUE THREAD

**Issue by eburgueno** [trinityrnaseq/trinityrnaseq#190]
One of our scientists has been having issues running Trinity, sometimes even with the test data provided. In all cases we were able to reproduce the error albeit non-consistently. Trinity fails at the fastool step with a message similar to this one:

```
Thread 1 terminated abnormally: Error, cmd: /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/19_C9WECANXX_CTGAAGCT-ATAGAGGC_L006_R1.fastq.gz) >> left.fa died with ret 256 at /home/<user>/trinityrnaseq-2.3.2/util/insilico_read_normalization.pl line 769.
```

Sometimes the issue is with `left.fa`, sometimes with `right.fa`. In all cases we see something similar to the following:

```
CMD: /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/19_C9WECANXX_CTGAAGCT-ATAGAGGC_L006_R2.fastq.gz) >> right.fa
Sequences parsed: 307
fastool: parsing error, truncated sequence and/or quality detected!
```

Running fastool directly multiple times, I get inconsistent results each run:

```
$ mkdir /tmp/a; cd /tmp/a

$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 307
fastool: parsing error, truncated sequence and/or quality detected!
$ ll
total 52
-rw-r--r--. 1 root root 51741 Nov 29 10:38 left.fa
$ rm left.fa 
rm: remove regular file ‘left.fa’? y
$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 11761
$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 4305
$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 25733
$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 49013
$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 2255
$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 45684
$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 2870
$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta <(gunzip -c /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz) >> left.fa
Sequences parsed: 26652
```

If I uncompress the fastq file, fastool works as expected:

```
$ zcat /<path>/18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq.gz > 18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq

$ /home/<user>/trinityrnaseq-2.3.2/util/..//trinity-plugins/fastool/fastool --illumina-trinity --to-fasta 18_C9WECANXX_GAATTCGT-ATAGAGGC_L005_R1.fastq >> left.fa
Sequences parsed: 5381400
```

**Comment by brianjohnhaas** [comment]
I'll look into changing the syntax to make it more robust.  

-Brian
(by iPhone)

**Comment by AlesBucek** [comment]
I experience probably identical or very similar problem with Trinity during the first step - i.e. parsing the raw reads. 
First, I run succesfully Trinity with *fastq.gz files which were previously trimmed and quality-filtered by Trimmomatic.

`Trinity --seqType fq --left left_Trimmed.fastq.gz --right right_Trimmed.fastq.gz --CPU 4 --max_memory 39G --output Trinity_Trimmed > "Trinity_Trimmed.log" 2>&1`

It gave reasonable number of transcripts and in the logfile I could check that all the reads from the *fastq.gz files were parsed durig the initial steps of Trinity.

But when I wanted to check how will Trinity deal with non-trimmed reads I noticed extremely low number of contigs and when I went through the logfile I see than only small portion or reads (both left and right) were parsed (information in Trinity output in line "Sequences parsed:"). 

I rerun Trinity cca 10 times with the untrimmed reads and each time there was a different number of parsed reads (but always just a fraction of reads from the input fastq.gz files so the trinity resulted in unreasonably small number of transcripts) and in all but one cases trinity run without reporting any errors. Only once, trinity failed, reporting error:

`fastool: parsing error, truncated sequence and/or quality detected!`
